src/components/ListSection.tsx

```diff
@@ -1,3 +1,52 @@
+import { useEffect, useState } from 'react';
+import { instance } from '../libs/api';
+
+// records 데이터를 받아오기 위한 Records interface 생성
+interface Records {
+  id: string;
+  fields: Fields;
+}
+
+// fields object를 위한 interface 생성
+interface Fields {
+  name: string;
+}
+
 export default function ListSection() {
-  return <section></section>;
+  const [list, setList] = useState<Records[]>([]);
+
+  // ComponentDidMount(=useEffect) => API 호출
+
+  // 1. useEffect Hook 내부에서 정의
+  // 컴포넌트가 마운트 되면, getRes 함수가 호출됨
+  // getRes함수에서 res함수는 API를 호출(/todo)
+  // setter인 setTodo는 getter인 todo에게 res함수에 담긴 데이터를 건내줌
+
+  // useEffect(() => {
+  //   const getRes = async () => {
+  //     const res = await instance.get('/todo');
+  //     setTodo(res.data.records);
+  //   };
+  //   getRes();
+  // }, []);
+
+  // 2. useEffect Hook 외부에서 정의
+  const getRes = async () => {
+    const res = await instance.get('/todo');
+    setList(res.data.records);
+  };
+  useEffect(() => {
+    getRes();
+  }, []);
+
+  return (
+    <section>
+      {list.map(list => {
+        return <li key={list.id}>{list.fields.name}</li>;
+      })}
```

Comment:
필수는 아니고, 취향에 따라 갈리는 부분이지만

```suggestion
      {list.map(list => (
        <li key={list.id}>{list.fields.name}</li>;
      ))}
```

이렇게 둥근 괄호 `()`로 할 시 `return` 키워드가 축약되는 거 알고 계시면 좋을 거 같아요!